### PR TITLE
Add performance number - add reference - add evaluate json

### LINKED ISSUE #194: Add dice score for `spleen_deepedit_annotation`
There is no performance related descriptions in the readme file of the bundle `spleen_deepedit_annotation`, I think it is an useful information for users.
Hi @diazandr3s , could you please help to add this part based on the trained weights? Thanks in advance.
It may also need to supplement the `evaluate.json` (can refer to: https://github.com/Project-MONAI/model-zoo/blob/dev/models/spleen_ct_segmentation/configs/evaluate.json)

### PR DESCRIPTION
**Pull request by diazandr3s**
Signed-off-by: Andres <diazandr3s@gmail.com>

Fixes https://github.com/Project-MONAI/model-zoo/issues/194.

### CODE REVIEW

File: models/spleen_deepedit_annotation/configs/evaluate.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
+{
+    "imports": [
+        "$import glob",
+        "$import os",
+        "$import ignite"
+    ],
+    "bundle_root": "/workspace/tutorials/deepedit/deepedit_bundle",
+    "output_dir": "$@bundle_root + '/eval'",
+    "dataset_dir": "/workspace/Datasets/MSD_datasets/Task09_Spleen",
+    "datalist": "$list(sorted(glob.glob(@dataset_dir + '/imagesTs/*.nii.gz')))",
```
Comment (yiheng-wang-nv) [inline models/spleen_deepedit_annotation/configs/evaluate.json]: Thanks @diazandr3s for the quick updates. This json file seems the same as `inference.json`, and does not have any metric related settings. Here we need a file that can produce a validation score that is exactly the same as the readme file, thanks!

File: models/spleen_deepedit_annotation/docs/README.md [hunk 2]
```diff
@@ -21,6 +21,13 @@ Input: 3 channels CT image - one channel representing clicks for each segment (i
...
 
+## Model Performance
+
+The achieved Dice score on the validation set is:
+
+Spleen: 0.959
+
+
 ## commands example
 Execute training:
```
Comment (yiheng-wang-nv) [inline models/spleen_deepedit_annotation/docs/README.md:60]: here needs to provide a command to do the validation

Please also update the metadata.json for `changelog` and `version`, thanks!

File: models/spleen_deepedit_annotation/configs/evaluate.json (hunk 1 above)
Comment (Nic-Ma) [reply models/spleen_deepedit_annotation/configs/evaluate.json]: Hi @diazandr3s ,

Please refer to this one: https://github.com/Project-MONAI/model-zoo/blob/dev/models/spleen_ct_segmentation/configs/evaluate.json

Thanks.

Comment (diazandr3s) [reply models/spleen_deepedit_annotation/configs/evaluate.json]: Thanks for your comments, @yiheng-wang-nv @Nic-Ma 
I've just updated the PR according to your feedback

File: models/spleen_deepedit_annotation/docs/README.md [hunk 3]
```diff
@@ -21,6 +21,34 @@ Input: 3 channels CT image - one channel representing clicks for each segment (i
 
 Output: 2 channels: Label 1: spleen; Label 0: everything else - This depends on the dictionary "label_names" defined in train.json and inference.json ("label_names": {"spleen": 1, "background": 0},)
 
+## Model Performance
+
+The achieved Dice score on the validation set is:
+
+Spleen: 0.971
```
Comment (yiheng-wang-nv) [inline models/spleen_deepedit_annotation/docs/README.md]: Hi @diazandr3s , thanks for the updates, I still cannot reproduce this number. The following is the log:
```
2022-11-14 03:17:04,180 - INFO - --- input summary of monai.bundle.scripts.run ---
2022-11-14 03:17:04,180 - INFO - > runner_id: 'evaluating'
2022-11-14 03:17:04,180 - INFO - > meta_file: 'configs/metadata.json'
2022-11-14 03:17:04,180 - INFO - > config_file: ['configs/train.json', 'configs/evaluate.json']
2022-11-14 03:17:04,180 - INFO - > logging_file: 'configs/logging.conf'
2022-11-14 03:17:04,180 - INFO - > bundle_root: '.'
2022-11-14 03:17:04,181 - INFO - > dataset_dir: '/workspace/data/medical/Task09_Spleen/'
2022-11-14 03:17:04,181 - INFO - ---


2022-11-14 03:17:04,181 - INFO - set logging properties based on config: configs/logging.conf.
there are more than 1 component have name `Interaction`: ['monai.apps.deepedit.interaction', 'monai.apps.deepgrow.interaction'], use the first one `monai.apps.deepedit.interaction. if want to use others, please set its full module path in `_target_` direc
tly.
2022-11-14 03:17:05,428 - ignite.engine.engine.SupervisedEvaluator - INFO - Engine run resuming from iteration 0, epoch 0 until 1 epochs
2022-11-14 03:17:05,501 - ignite.engine.engine.SupervisedEvaluator - INFO - Restored all variables from ./models/model.pt
2022-11-14 03:17:10,964 - INFO - Number of simulated clicks: 5
2022-11-14 03:17:12,654 - INFO - Number of simulated clicks: 6
2022-11-14 03:17:15,416 - INFO - Number of simulated clicks: 2
2022-11-14 03:17:17,321 - INFO - Number of simulated clicks: 5
2022-11-14 03:17:19,385 - INFO - Number of simulated clicks: 5
2022-11-14 03:17:21,137 - INFO - Number of simulated clicks: 5
2022-11-14 03:17:22,800 - INFO - Number of simulated clicks: 5
2022-11-14 03:17:24,256 - INFO - Number of simulated clicks: 3
2022-11-14 03:17:25,732 - INFO - Number of simulated clicks: 3
2022-11-14 03:17:26,005 - ignite.engine.engine.SupervisedEvaluator - INFO - Got new best metric of val_mean_dice: 0.9151045680046082
2022-11-14 03:17:26,006 - ignite.engine.engine.SupervisedEvaluator - INFO - Epoch[1] Metrics -- val_mean_dice: 0.9151
2022-11-14 03:17:26,006 - ignite.engine.engine.SupervisedEvaluator - INFO - Key metric: val_mean_dice best value: 0.9151045680046082 at epoch: 1
Mean of empty slice
All-NaN slice encountered
All-NaN slice encountered
All-NaN slice encountered
Degrees of freedom <= 0 for slice.
2022-11-14 03:17:26,010 - ignite.engine.engine.SupervisedEvaluator - INFO - Epoch[1] Complete. Time taken: 00:00:20.508
2022-11-14 03:17:26,010 - ignite.engine.engine.SupervisedEvaluator - INFO - Engine run complete. Time taken: 00:00:20.582
[null, null]
```
It got 0.9151 mean dice. In addition, I run the evaluating command for 3 times, and produced 3 different mean dice scores. I'm not familiar with the deepedit method, is it correct that the deepedit method cannot produce deterministic results?

File: models/spleen_deepedit_annotation/docs/README.md [hunk 4]
```diff
@@ -21,6 +21,34 @@ Input: 3 channels CT image - one channel representing clicks for each segment (i
...
+Command to execute:
+
+```
+python -m monai.bundle run evaluating --meta_file configs/metadata.json --config_file "['configs/train.json','configs/evaluate.json']" --logging_file configs/logging.conf
+```
+
+### Train Dice
+
+![Train Dice](./train_dice_deepedit.png)
+
```
Comment (yiheng-wang-nv) [inline models/spleen_deepedit_annotation/docs/README.md:42]: `max_epochs` in `train.json` is 500, but the horizontal axis in the figure is 90

Comment (diazandr3s) [reply models/spleen_deepedit_annotation/docs/README.md:42]: I didn't train the model for all the 500 epochs. Is that an issue? I just wanted to show the plots I've got during training

File: models/spleen_deepedit_annotation/docs/README.md (hunk 3 above)
Comment (diazandr3s) [reply models/spleen_deepedit_annotation/docs/README.md]: This depends on the number of simulated clicks. DeepEdit is both an automatic and interactive method. It performs better if the user provides clicks or more clicks are simulated during validation.

File: models/spleen_deepedit_annotation/docs/README.md (hunk 4 above)
Comment (yiheng-wang-nv) [reply models/spleen_deepedit_annotation/docs/README.md:42]: I'm not sure if it is a hard requirement, but if there is no need to train for 500 epochs, maybe reduce the number in the config file is better (for test and for users to reproduce)?

Hi @Nic-Ma @wyli @tangy5 , what do you think?

File: models/spleen_deepedit_annotation/docs/README.md (hunk 3 above)
Comment (diazandr3s) [reply models/spleen_deepedit_annotation/docs/README.md]: I added the reference to the DeepEdit paper so users know more about this method.

File: models/spleen_deepedit_annotation/docs/README.md (hunk 4 above)
Comment (diazandr3s) [reply models/spleen_deepedit_annotation/docs/README.md:42]: I can change the default number of epochs. Perhaps for spleen segmentation, it is not needed the 500 epochs. But users may want to use this method for other more complex tasks. Let me know :)

File: models/spleen_deepedit_annotation/docs/README.md (hunk 3 above)
Comment (yiheng-wang-nv) [reply models/spleen_deepedit_annotation/docs/README.md]: thanks for the explain, just to make sure if the weights file is align with the readme file.
In the user point of view (as what I checked here), they may care about how to make use of the weights and reach to 0.971. Could you please help to add more descriptions about how to reach to this number (like provide clicks as you said)? and if can not get the deterministic result, may need to write sth like `around 0.971`?

File: models/spleen_deepedit_annotation/docs/README.md (hunk 4 above)
Comment (yiheng-wang-nv) [reply models/spleen_deepedit_annotation/docs/README.md:42]: It makes sense @diazandr3s , thanks! In this way, maybe we can add a sentence for the figure like "A graph showing the training loss/metric over 90 epochs." (like what we do in the spleen ct segmentation bundle)

Comment (diazandr3s) [reply models/spleen_deepedit_annotation/docs/README.md:42]: Makes sense. Just added some comments for this and the Dice score. I think writing a single number doesn't make sense. I commented on that as well.

**Comment by yiheng-wang-nv** [comment]
Hi @diazandr3s , thanks for the updates. I just asked @zephyrie for help to upload the figures you added, and after that you can use urls into the readme file directly (like https://developer.download.nvidia.com/assets/Clara/Images/monai_swin_unetr_segmentation_val.png for swin unetr bundle). After this step I think the PR can be merged.

**Comment by diazandr3s** [comment]
> Hi @diazandr3s , thanks for the updates. I just asked @zephyrie for help to upload the figures you added, and after that you can use urls into the readme file directly (like https://developer.download.nvidia.com/assets/Clara/Images/monai_swin_unetr_segmentation_val.png for swin unetr bundle). After this step I think the PR can be merged.

Please let me know whenever you get the URLs.
Thanks! :)

**Comment by yiheng-wang-nv** [comment]
Hi @diazandr3s , the figures are:

https://developer.download.nvidia.com/assets/Clara/Images/monai_spleen_deepedit_annotation_train_dice.png
https://developer.download.nvidia.com/assets/Clara/Images/monai_spleen_deepedit_annotation_val_dice.png
https://developer.download.nvidia.com/assets/Clara/Images/monai_spleen_deepedit_annotation_train_loss.png

**Comment by diazandr3s** [comment]
> Hi @diazandr3s , the figures are:
> 
> https://developer.download.nvidia.com/assets/Clara/Images/monai_spleen_deepedit_annotation_train_dice.png https://developer.download.nvidia.com/assets/Clara/Images/monai_spleen_deepedit_annotation_val_dice.png https://developer.download.nvidia.com/assets/Clara/Images/monai_spleen_deepedit_annotation_train_loss.png

Thanks, @yiheng-wang-nv 
Just updated the readme file accordingly.

**Review by yiheng-wang-nv — APPROVED** [review]

**Review by yiheng-wang-nv — APPROVED** [review]

**Review by Nic-Ma — APPROVED** [review]
